L1Trigger/L1TZDC/plugins/L1TZDCProducer.cc

```diff
@@ -0,0 +1,259 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "L1Trigger/L1TCalorimeter/interface/Stage2Layer2FirmwareFactory.h"
+#include "L1Trigger/L1TCalorimeter/interface/Stage2MainProcessor.h"
+#include "L1Trigger/L1TCalorimeter/interface/CaloTools.h"
+
+#include "L1Trigger/L1TCalorimeter/interface/CaloParamsHelper.h"
+#include "CondFormats/DataRecord/interface/L1TCaloParamsRcd.h"
+#include "CondFormats/DataRecord/interface/L1TCaloParamsO2ORcd.h"
+
+#include "DataFormats/L1TCalorimeter/interface/CaloTower.h"
+#include "DataFormats/L1Trigger/interface/EGamma.h"
+#include "DataFormats/L1Trigger/interface/Tau.h"
+#include "DataFormats/L1Trigger/interface/Jet.h"
+#include "DataFormats/L1Trigger/interface/EtSum.h"
+
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+
+//
+// class declaration
+//
+
+using namespace l1t;
+
+class L1TZDCProducer : public edm::stream::EDProducer<> {
+public:
+  explicit L1TZDCProducer(const edm::ParameterSet& ps);
+  ~L1TZDCProducer() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  //  void endRun(edm::Run const&, edm::EventSetup const&) override;
+
+  int zdcLUTIndexHelper(int iDetPos, int iBXPos);
+
+  // ----------member data ---------------------------
+
+  // input tokens
+  //Add the ZDC token, candidateToken for caloParamsHelper
+  edm::EDGetTokenT<QIE10DigiCollection> zdcToken_;
+  edm::ESGetToken<CaloParams, L1TCaloParamsRcd> candidateToken_;
+
+  //input ints
+  int bxFirst_;
+  int bxLast_;
+  int sampleToCenterBX_;
+
+  // put tokens
+  edm::EDPutTokenT<EtSumBxCollection> etToken_;
+
+  //Following the L1TStage2Layer2Producer
+  std::unique_ptr<CaloParamsHelper> params_;
+  int scaleFactor_;
+};
+
+L1TZDCProducer::L1TZDCProducer(const edm::ParameterSet& ps) {
+  // register what you produce
+  etToken_ = produces<EtSumBxCollection>();
+
+  // register what you consume and keep token for later access:
+  zdcToken_ = consumes<QIE10DigiCollection>(ps.getParameter<edm::InputTag>("zdcDigis"));
+  candidateToken_ = esConsumes<CaloParams, L1TCaloParamsRcd, edm::Transition::BeginRun>();
+
+  bxFirst_ = ps.getParameter<int>("bxFirst");
+  bxLast_ = ps.getParameter<int>("bxLast");
+  sampleToCenterBX_ = ps.getParameter<int>("sampleToCenterBX");
+}
+
+// ------------ method called to produce the data  ------------
+void L1TZDCProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+  using namespace l1t;
+
+  LogDebug("l1t|stage 2") << "L1TZDCProducer::produce function called..." << std::endl;
```

Comment:
```suggestion
  LogDebug("L1TZDCProducer") << "L1TZDCProducer::produce function called..." << std::endl;
```
I would naively use `"L1TZDCProducer"` as you do in the rest of the plugin.

Reply:
You are correct, this is fixed in latest commit